File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        count = dict()
        for elem in nums:
            if elem in count.keys(): count[elem] += 1
            else: count[elem] = 1
        k=0
        m=0
        for key, value in count.items():
            if value>m:
                m=value
                k=key
        return k
        

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let counter = 0
    let current = 0
    for (let i = 0; i < nums.length ; i++){
        if (counter === 0){
            current = nums[i]
            counter += 1
        }else{
            if(current === nums[i]){
                counter += 1
            }else{
                counter -= 1
            }
        }
    }
    return current;
};